lizard_auth_server/models.py

```diff
@@ -746,3 +740,136 @@ def as_dict(self):
+        verbose_name=_('user'),
+        related_name='profile')
+    company = models.ForeignKey(
+        'Company',
+        verbose_name=_('company'),
+        blank=True,
+        null=True)
+    created_at = models.DateTimeField(
+        verbose_name=_('created on'),
+        auto_now_add=True)
+    updated_at = models.DateTimeField(
+        verbose_name=_('updated on'),
+        auto_now=True)
+
+    def __str__(self):
+        return self.user.username
+
+    class Meta:
+        verbose_name = _("user profile")
+        verbose_name_plural = _("user profiles")
+
+    @property
+    def username(self):
+        return self.user.username
+
+    @property
+    def full_name(self):
+        return self.user.get_full_name()
+
+    @property
+    def first_name(self):
+        return self.user.first_name
+
+    @property
+    def last_name(self):
+        return self.user.last_name
+
+    @property
+    def email(self):
+        return self.user.email
+
+
+class Company(models.Model):
+    """Replacement for Organisation."""
+    name = models.CharField(
+        verbose_name=_('name'),
+        max_length=255,
+        null=False,
+        blank=False,
+        unique=True)
+    unique_id = models.CharField(
+        verbose_name=_('unique id'),
+        max_length=32,
+        unique=True,
+        default=create_new_uuid)
+    guests = models.ManyToManyField(
+        'Profile',
+        related_name='guest_companies',
+        verbose_name=_('guests'),
+        blank=True,
+        help_text=_("Guests or external users."))
+    administrators = models.ManyToManyField(
+        'Profile',
+        related_name='admin_companies',
+        verbose_name=_('administrators'),
+        blank=True,
+        help_text=_("Can manage other user profiles."))
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        verbose_name = _('company')
+        verbose_name_plural = _('companies')
+
+
+class Site(models.Model):
+    """Replacement for Portal."""
+    name = models.CharField(
+        verbose_name=_('name'),
+        max_length=255,
+        null=False,
+        blank=False,
+        help_text=_('Name used to refer to this site.'))
+    available_to = models.ManyToManyField(
+        'Company',
+        related_name='sites',
+        verbose_name=_('available to'),
+        blank=True)
+    sso_secret = models.CharField(
+        verbose_name=_('shared secret'),
+        max_length=64,
+        unique=True,
+        default=GenKey('Portal', 'sso_secret'),
+        help_text=_('Secret shared between SSO client and '
+                    'server to sign/encrypt communication.'))
+    sso_key = models.CharField(
+        verbose_name=_('identifying key'),
+        max_length=64,
+        unique=True,
+        default=GenKey('Portal', 'sso_key'),
+        help_text=_('String used to identify the SSO client.'))
+    allowed_domain = models.CharField(
+        verbose_name=_('allowed domain(s)'),
+        max_length=255,
```

Comment:
Als er meerdere domeinen toegestaan moeten worden zoals "gemeentegroningen.staging.lizard.net" en "vaagprojectinhetbuitenland.ietsinternationaals.stichting.org", kunnen we dan niet beter de max length wat opkrikken? 1000 karakters ofzo?